From 875c169a681b8fdd0065dc3b28f10132b908d027 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Aur=C3=A9lien=20Jacobs?= <aurel@gnuage.org>
Date: Sun, 12 Jul 2026 23:07:20 +0100
Subject: [PATCH] Fix an inverted logic issue for the force parameter of the
 load command

The bug was introduced in the following commit:
83537e2  Tweak handling of parameters.
---
 fake-hwclock | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fake-hwclock b/fake-hwclock
index 641c49f..d83afc8 100755
--- a/fake-hwclock
+++ b/fake-hwclock
@@ -56,7 +56,7 @@ case $COMMAND in
             SAVED="$(cat $FILE)"
             SAVED_SEC=$(date -u -d "$SAVED" '+%s')
             NOW_SEC=$(date -u '+%s')
-            if [ "$FORCE"x = "false"x ] || [ $NOW_SEC -le $SAVED_SEC ] ; then
+            if [ "$FORCE"x != "false"x ] || [ $NOW_SEC -le $SAVED_SEC ] ; then
                 date -u -s "$SAVED"
             else
                 echo "Current system time: $(date -u '+%Y-%m-%d %H:%M:%S')"
-- 
2.43.0

